instrumentation/async-http-client/async-http-client-1.9/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/asynchttpclient/v1_9/AsyncHttpClientSingletons.java

```diff
@@ -5,27 +5,37 @@
 
 package io.opentelemetry.javaagent.instrumentation.asynchttpclient.v1_9;
 
+import com.ning.http.client.AsyncHandler;
 import com.ning.http.client.Request;
 import com.ning.http.client.Response;
 import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.util.VirtualField;
 import io.opentelemetry.javaagent.bootstrap.internal.JavaagentHttpClientInstrumenters;
 
 public final class AsyncHttpClientSingletons {
   private static final String INSTRUMENTATION_NAME = "io.opentelemetry.async-http-client-1.9";
 
   private static final Instrumenter<Request, Response> INSTRUMENTER;
 
+  private static final VirtualField<AsyncHandler<?>, AsyncHandlerData> VIRTUAL_FIELD;
+
   static {
     INSTRUMENTER =
         JavaagentHttpClientInstrumenters.create(
             INSTRUMENTATION_NAME,
             new AsyncHttpClientHttpAttributesGetter(),
             HttpHeaderSetter.INSTANCE);
+
+    VIRTUAL_FIELD = VirtualField.find(AsyncHandler.class, AsyncHandlerData.class);
   }
 
   public static Instrumenter<Request, Response> instrumenter() {
     return INSTRUMENTER;
   }
 
+  public static VirtualField<AsyncHandler<?>, AsyncHandlerData> virtualField() {
```

Comment:
Why use a accessor method instead of making `VIRTUAL_FIELD` public? I think accessing a public constant is more readable, as `virtualField()` could have unknown side effects, while a constant access does not.